ring/basic_lifecycler_test.go

```diff
@@ -323,9 +327,12 @@ func TestBasicLifecycler_HeartbeatAfterBackendRest(t *testing.T) {
 			desc.GetTimestamp() > 0 &&
 			desc.GetState() == ACTIVE &&
 			Tokens(desc.GetTokens()).Equals(registerTokens) &&
-			desc.GetAddr() == cfg.Addr &&
-			desc.GetRegisteredAt().Unix() == expectedRegisteredAt.Unix()
+			desc.GetAddr() == cfg.Addr
 	})
+
+	// Ensure the registration timestamp has been updated.
+	desc, _ := getInstanceFromStore(t, store, testInstanceID)
+	assert.Greater(t, desc.GetRegisteredTimestamp(), prevRegisteredAt.Unix())
```

Comment:
We should check that `lifecycler.GetRegisteredAt()` is also correct.

Reply:
Done!